src/lib/hooks/useImageSource.js

```diff
@@ -1,47 +1,72 @@
-import { useEffect, useMemo } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 import { first, last } from 'lodash'
 
-import { isGoodDollarImage, isValidLocalImage, isValidRootImage } from '../utils/image'
+import avatarStorage from '../gundb/UserAvatarStorage'
+import { isValidCID } from '../utils/ipfs'
+import { asDataUrl, getBase64Source, isGoodDollarImage, isImageRecord, isValidBase64Image } from '../utils/image'
+
 import logger from '../logger/pino-logger'
-import useProfileAvatar from './useProfileAvatar'
 
 const log = logger.child({ from: 'useImageSource' })
+const emptySource = [false, null]
+
+const imageToSource = source => {
+  // checking is image record or base64 data url
+  const isRecord = isImageRecord(source)
+
+  // if not return null
+  if (!isRecord && !isValidBase64Image(source)) {
+    return null
+  }
+
+  let base64 = source
+
+  if (isRecord) {
+    // assemble data url from image record
+    base64 = asDataUrl(source)
+  }
+
+  return [false, getBase64Source(base64)]
+}
 
-export default (source, skipCache = false) => {
+export default (source, size = 'small') => {
```

Comment:
hmm seems unnecessarily complicated. i've pushed a refactoring

Reply:
will check it